cmd/samples/recipes/helloworld/replay_test.go

```diff
@@ -32,6 +32,10 @@ func (s *replayTestSuite) TearDownTest() {
 	s.mockCtrl.Finish() // assert mock’s expectations
 }
 
+// This replay test is the recommended way to make sure changing workflow code is backward compatible without non-deterministic errors.
+// "helloworld.json" can be downloaded from cadence CLI:
+//      cadence --do samples-domain wf show -w helloworld_d002cd3a-aeee-4a11-aa30-1c62385b4d87 --output_filename ~/tmp/helloworld.json
+// Or from Cadence Web UI. And you may need to change workflowType in the first event.
```

Comment:
Should we instead fix the replay to get the workflowType from the first event?

Reply:
Yes, I created an issue for it: https://github.com/uber-go/cadence-client/issues/721